### PR TITLE
Possible new version

### PR DESCRIPTION
**Pull request by vianamp**
Encapsulates everything in a single loop. The additional loop for
applying gzip was left out for sake of clarity. The code will work fine
on Mac OS, but it may skip the first folder in other systems that do
have the `.DS_Store folder`. The code works for images of any size
smaller than 256x256, but keep in mind images of size 28x28 are
required for MNIST-based machine learning algorithms.

### CODE REVIEW

**Comment by gskielian** [comment]
Merging Codebase, thanks! : ) 

Looking forward to trying this out on some new 28x28 MNIST size images, btw what is your choice tool for  MNIST neural-net training?

**Comment by vianamp** [comment]
I am mostly doing what is described in this tutorial:

[https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/docker/notebooks/3_mnist_from_scratch.ipynb](url)

> The raw input passes through several convolution and max pooling layers with rectified linear activations before several fully connected layers and a softmax loss for predicting the output class. During training, we use dropout.

How are you approaching this problem? Have you ever tried to work on images bigger than 28x28?
